frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -362,20 +362,17 @@ class MidCircuitMeasure(HybridOp):
     binder = qmeasure_p.bind
 
     def __init__(self, *args, **kwargs):
-        self.bypass_postselect = kwargs.pop("bypass_postselect", False)
-        self.postselect = kwargs.pop("postselect", None)
-        self.reset = kwargs.pop("reset", False)
         super().__init__(*args, **kwargs)
+        self.postselect = self.in_classical_tracers[-2]
+        self.reset = self.in_classical_tracers[-1]
 
     def trace_quantum(self, ctx, device, trace, qrp) -> QRegPromise:
         op = self
         wire = op.in_classical_tracers[0]
         qubit = qrp.extract([wire])[0]
-        if self.bypass_postselect:
-            qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)
-        else:
-            postselect = op.in_classical_tracers[1]
-            qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=postselect)
+        qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit)
+        # TODO: execute post-selection depending on qnode config
+        # qubit2 = op.bind_overwrite_classical_tracers(ctx, trace, qubit, postselect=op.postselect)
```

Comment:
With this change, we're effectively only supporting `postselect_mode="hw-like"` in the interim right? I'm fine with either only supporting `hw-like` or `fill-shots` for now so nothing needs to be changed.

Reply:
Correct, this is hardcoding `postselect_mode="hw-like"`.

Reply:
The second option (`fill-shots`) now fails within `dynamic_one_shot`. I would suggest merging as is with a couple xfailing tests. When the QNode config comes in, we'll be able to un-xfail those passing `single-branch-statistics` which is their current execution mode. Then we can fix `fill-shots` in a follow-up PR. Sounds good to you both @mudit2812 @dime10 ?